src/lib/templates.js

```diff
@@ -2,115 +2,132 @@ const prefix = 'docs-searchbar';
 const suggestionPrefix = `${prefix}-suggestion`;
 const footerPrefix = `${prefix}-footer`;
 
-const templates = {
-  suggestion: `
-  <a class="${suggestionPrefix}
-    {{#isCategoryHeader}}${suggestionPrefix}__main{{/isCategoryHeader}}
-    {{#isSubCategoryHeader}}${suggestionPrefix}__secondary{{/isSubCategoryHeader}}
-    "
-    aria-label="Link to the result"
-    href="{{{url}}}"
-    >
-    <div class="${suggestionPrefix}--category-header">
-        <span class="${suggestionPrefix}--category-header-lvl0">{{{category}}}</span>
-    </div>
-    <div class="${suggestionPrefix}--wrapper">
-      <div class="${suggestionPrefix}--subcategory-column">
-        <span class="${suggestionPrefix}--subcategory-column-text">{{{subcategory}}}</span>
+let suggestion;
+let suggestionSimple;
+let footer;
+let empty;
+let searchBox;
+
+function templates(
+  options = {
+    suggestion,
+    suggestionSimple,
+    footer,
+    empty,
+    searchBox,
+  }
```

Comment:
```suggestion
  options = {}
```
Because: 
```javascript
// this value 
options = {
   suggestion,
   suggestionSimple,
   footer,
   empty,
   searchBox
}
```

when assigned with this
```javascript
Object.assign({}, defaults, options);
```

will have as output the following:
```javascript
options = {
   suggestion: undefined,
   suggestionSimple: undefined,
   footer: undefined,
   empty: undefined,
   searchBox: undefined
}
```

Instead of the default template provided.

You can try it out in the code by replacing 
```javascript
// this.autocompleteOptions.templates === undefined
this.templates = new Templates(this.autocompleteOptions.templates || {} );
```
with
```javascript
// this.autocompleteOptions.templates === undefined
this.templates = new Templates(this.autocompleteOptions.templates);
```

Which shouldn't; be useful with a default value in the Template function.